rust/src/core.rs

```diff
@@ -174,4 +184,33 @@ mod test {
         }
         Ok(())
     }
+
+    #[test]
+    fn test_is_container_image_reference() -> Result<()> {
+        use super::is_container_image_reference;
+
+        let refspec_type_checksum =
+            "ee10f8e7ef638d78ba9a9596665067f58021624118875cc4079568da6c63efb0";
+        assert!(!is_container_image_reference(refspec_type_checksum));
+
+        let refspec_type_ostree_with_remote = "fedora:fedora/x86_64/coreos/testing-devel";
+        assert!(!is_container_image_reference(
+            refspec_type_ostree_with_remote
+        ));
+        let refspec_type_ostree = "fedora/x86_64/coreos/foo-branch";
+        assert!(!is_container_image_reference(refspec_type_ostree));
+
+        let refspec_type_container = "containers-storage:localhost/fcos:latest";
+        assert!(is_container_image_reference(refspec_type_container));
+        let refspec_type_container = "docker://quay.io/test-repository/os:version1";
+        assert!(is_container_image_reference(refspec_type_container));
+        let refspec_type_container = "registry:docker.io/test-repository/os:latest";
+        assert!(is_container_image_reference(refspec_type_container));
+        let refspec_type_container = "registry:customhostname.com:8080/test-repository/os:latest";
+        assert!(is_container_image_reference(refspec_type_container));
+        let refspec_type_container = "docker://quay.io/test-repository/os@sha256:6006dca86c2dc549c123ff4f1dcbe60105fb05886531c93a3351ebe81dbe772f";
+        assert!(is_container_image_reference(refspec_type_container));
```

Comment:
Nice, thanks for adding unit tests!

See e.g. https://github.com/ostreedev/ostree-rs-ext/blob/1d0b14db6d3104c5d08a4c99450c0d676fd4a579/lib/src/container/mod.rs#L146 which is similar but deduplicates the test code a bit by having an array of "valid things" and then just iterates over it.